docs/machine_learning.rst

```diff
@@ -89,30 +89,36 @@ Finally execute this whole training flow as a batch job::
     training_job = model.create_job()
     training_job.start_and_wait()
 
-
 Inference
 ----------
 
 When the batch job finishes successfully, the trained model can then be used
 with the ``predict_random_forest`` process on the raster data cube
 (or another cube with the same band structure) to classify all the pixels.
 
+Thus, the training job results is inspected to fetch the trained models url::
+
+    from openeo.rest.job import JobResults
+
+    results : JobResults = training_job.get_results()
+    links = results.get_metadata()['links']
+    ml_model_metadata_url = [link for link in links if 'ml_model_metadata.json' in link['href']][0]['href']
+    print(ml_model_metadata_url)
+
+
+Next, load the model from the URL::
+
+    from openeo.rest.mlmodel import MlModel
+    saved_model = MlModel.load_ml_model(connection=connection, id=ml_model_metadata_url)
+
 Technically, the openEO ``predict_random_forest`` process has to be used as a reducer function
 inside a ``reduce_dimension`` call, but the openEO Python client library makes it
 a bit easier by providing a :py:meth:`~openeo.rest.datacube.DataCube.predict_random_forest` method
 directly on the :py:class:`~openeo.rest.datacube.DataCube` class, so that you can just do::
 
     predicted = cube.predict_random_forest(
-        model=training_job.job_id,
+        model = saved_model,
         dimension="bands"
     )
 
     predicted.download("predicted.GTiff")
-
-
-We specified the model here by batch job id (string),
-but it can also be specified in other ways:
-as :py:class:`~openeo.rest.job.BatchJob` instance,
-as URL to the corresponding STAC Item that implements the `ml-model` extension,
-or as :py:class:`~openeo.rest.mlmodel.MlModel` instance (e.g. loaded through
-:py:meth:`~openeo.rest.connection.Connection.load_ml_model`).
```

Comment:
Why are the docs about the other options (batch job id, batch job instance, ...) removed? Is it because this is not valid anymore? If so, then the docs in a couple of places should be updated too

Reply:
It was indeed confusion on my part since the other methods, weren't working for me in the CDSE backends.